diff --git a/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/CMakeLists.txt b/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/CMakeLists.txt
index 09bd8cdfb..707be3f59 100644
--- a/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/CMakeLists.txt
+++ b/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/CMakeLists.txt
@@ -31,6 +31,7 @@ iree_cc_library(
     "ConvertConv2DToIm2ColOp.cpp"
     "ConvertConv2DToWinograd.cpp"
     "ConvertToLoops.cpp"
+    "Guacamole.cpp"
     "DecomposeAttention.cpp"
     "DecomposeIm2col.cpp"
     "DecomposeWinogradPass.cpp"
diff --git a/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/Passes.td b/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/Passes.td
index 6454c5ab0..803d5836b 100644
--- a/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/Passes.td
+++ b/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/Passes.td
@@ -9,6 +9,20 @@
 
 include "mlir/Pass/PassBase.td"
 
+def GuacamolePass :
+    Pass<"guacamole", ""> {
+  let summary = "IREE hello world pass at linalg level";
+  let description = [{
+    Counts all instructions and function names.
+  }];
+  let statistics = [
+    Statistic<"funcCount", "func-count",
+              "how many functions are in the source file?">,
+    Statistic<"insCount", "ins-func",
+              "how many instructions are in the source file?">,
+  ];
+}
+
 def LinalgExtToLoopsPass :
     InterfacePass<"iree-linalg-ext-to-loops", "mlir::FunctionOpInterface"> {
   let summary = "Convert LinalgExt ops to loops and Linalg ops.";
